swarm/network/simulation/node.go

```diff
@@ -256,8 +257,14 @@ func (s *Simulation) UploadSnapshot(snapshotFile string, opts ...AddNodeOption)
 	if err != nil {
 		return err
 	}
-	log.Info("Snapshot loaded")
-	return nil
+
+	err = s.WaitTillSnapshotRecreated(ctx, &snap)
+	if err == nil {
```

Comment:
Please, inline the error wherever possible. => `if err := f(); err != nil`
By keeping the variable scope minimal you don't pollute the surrounding scope and you make it clear I don't have to worry about/understand the code what comes later, I can just focus the if statement. => lower cognitive load

Reply:
@frncmx I don't think we should stick every assignment in an `if` statement. The accustomed syntax is to do the assignment then sequentially check for `err != nil`, I find this more readable somehow. That being said, checking for `err == nil` is something I'd like to avoid.

Reply:
I agree that `if err != nil` and early return should be preferred. Good point!